File: 203. Remove Linked List Elements/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
// 1663554347
class Solution{
public:
    ListNode* removeElements(ListNode *head,int val){
        if(!head) return head;
        ListNode *result = new ListNode{head->val,nullptr};
        ListNode *current1 = head->next;
        ListNode *current2 = result;
        while(current1 != nullptr){
            if(current1->val==val){current1 = current1->next;}
            else{
                current2->next = new ListNode{current1->val,nullptr};
                current1 = current1->next;
                current2 = current2->next;
            }
        }
        if(result->val==val) result = result->next;
        return result;
    }
};

File: 203. Remove Linked List Elements/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
# 1663593339
class Solution:
    def removeElements(self,head:Optional[ListNode],val: int)->Optional[ListNode]:
        if not head: return head
        while head and head.val==val:
            tmp = head
            head = head.next
            del tmp
        cur = head
        while cur and cur.next:
            if cur.next.val == val:
                tmp = cur.next
                cur.next = cur.next.next
                del tmp
            else:
                cur = cur.next
        return head                 
